catwalk/models/t5.py

```diff
@@ -21,31 +21,35 @@ def get_model(self) -> T5ForConditionalGeneration:
 
     def get_tokenizer(self) -> T5TokenizerFast:
         raise NotImplementedError
+    
+    def _predict_qa(self, task: Task, instances: Sequence[Dict[str, Any]], batch_size: int = 32) -> Iterator[Dict[str, Any]]:
+        qas = MappedSequence(task.instance_conversions[InstanceFormat.HF_QA], instances)
 
-    def predict(  # type: ignore
-        self,
-        task: Task,
-        instances: Sequence[Dict[str, Any]],
-        *,
-        batch_size: int = 32
-    ) -> Iterator[Dict[str, Any]]:
-        if not task.has_instance_conversion(InstanceFormat.T5_PROMPT):
-            raise UnsupportedTaskError(self, task)
+        model = self.get_model().eval()
+        tokenizer = self.get_tokenizer()
+        tokenizer.model_max_length = 1024 # We use this to be similar to the GPT truncation
```

Comment:
This has to come from the model or tokenizer though? What if I use a model that has a different max length? Or no max length, like Bloom?

Reply:
I believe T5 uses relative attention so it would support longer seq lengths so this is why I went with this fixed length. I think we can just use the 512 in the model config though as I anticipate that a longer default length will be memory consuming.